src/lib.rs

```diff
@@ -3830,6 +3815,54 @@ fn wait_on_child(cmd: &Command, program: &str, child: &mut Child) -> Result<(),
     }
 }
 
+/// Find the destination object path for each file in the input source files,
+/// and store them in the output Object.
+fn objects_from_files(files: &[Arc<Path>], dst: &Path) -> Result<Vec<Object>, Error> {
+    let mut objects = Vec::new();
+    for file in files.into_iter() {
+        let obj = if file.has_root() || file.components().any(|x| x == Component::ParentDir) {
+            // If `file` is an absolute path or might not be usable directly as a suffix due to
+            // using "..", use the `basename` prefixed with the `dirname`'s hash to ensure name
+            // uniqueness.
+            let basename = file
+                .file_name()
+                .ok_or_else(|| Error::new(ErrorKind::InvalidArgument, "file_name() failure"))?
+                .to_string_lossy();
+            let dirname = file
+                .parent()
+                .ok_or_else(|| Error::new(ErrorKind::InvalidArgument, "parent() failure"))?
+                .to_string_lossy();
+            let mut hasher = hash_map::DefaultHasher::new();
+            hasher.write(dirname.to_string().as_bytes());
+            dst.join(format!("{:016x}-{}", hasher.finish(), basename))
+                .with_extension("o")
+        } else {
+            dst.join(file).with_extension("o")
+        };
+        let obj = if !obj.starts_with(&dst) {
```

Comment:
Is this really necessary?

`obj` is created by joining `dst` with a name, so won't this always evaluates to `false`?

Reply:
I assume that this check is a remnant from before https://github.com/rust-lang/cc-rs/pull/684 was merged. Since there was no `file.has_root()` check before, the `dst.join(file)` could result in a path outside dst.

But yeah, it can probably be removed now.